spot_driver_cpp/src/interfaces/spot_interface.cpp

```diff
@@ -0,0 +1,394 @@
+  for (const auto& [child_frame_id, transform] : image_response.shot().transforms_snapshot().child_to_parent_edge_map())
+  {
+    // Do not publish static transforms for excluded frames
+    if (kExcludedStaticTfFrames.count(child_frame_id) > 0)
+    {
+      continue;
+    }
+
+    // Rename the parent link "arm0.link_wr1" to "link_wr1" as it appears in robot state
+    // which is used for publishing dynamic tfs elsewhere. Without this, the hand camera frame
+    // positions would never properly update as no other pipelines reference "arm0.link_wr1".
+    const auto parent_frame_id = (transform.parent_frame_name() == "arm0.link_wr1") ? "link_wr1" : transform.parent_frame_name();
+
+    geometry_msgs::msg::TransformStamped tform_msg;
+    tform_msg.header.stamp = applyClockSkew(image_response.shot().acquisition_time(), clock_skew);
+    tform_msg.header.frame_id = robot_name + "/" + parent_frame_id;
+    tform_msg.child_frame_id = robot_name + "/" + child_frame_id;
+    tform_msg.transform.translation.x = transform.parent_tform_child().position().x();
+    tform_msg.transform.translation.y = transform.parent_tform_child().position().y();
+    tform_msg.transform.translation.z = transform.parent_tform_child().position().z();
+    tform_msg.transform.rotation.w = transform.parent_tform_child().rotation().w();
+    tform_msg.transform.rotation.x = transform.parent_tform_child().rotation().x();
+    tform_msg.transform.rotation.y = transform.parent_tform_child().rotation().y();
+    tform_msg.transform.rotation.z = transform.parent_tform_child().rotation().z();
+    out.push_back(tform_msg);
+  }
+  return out;
+}
+}
+
+namespace spot_ros2
+{
+SpotInterface::SpotInterface()
+: client_sdk_{ ::bosdyn::client::CreateStandardSDK("get_image") }
+{
+}
+
+tl::expected<void, std::string> SpotInterface::createRobot(const std::string& ip_address, const std::string& robot_name)
+{
+  robot_name_ = robot_name;
+
+  auto create_robot_result = client_sdk_->CreateRobot(ip_address);
+  if(!create_robot_result.status)
+  {
+    return tl::make_unexpected("Received error result when creating SDK robot interface: " + create_robot_result.status.DebugString());
+  }
+
+  robot_ = std::move(create_robot_result.response);
+
+  return {};
+}
+
+tl::expected<void, std::string> SpotInterface::authenticate(const std::string& username, const std::string& password)
+{
+  if (!robot_)
+  {
+    return tl::make_unexpected("Spot SDK robot interface must be initialized before attempting to authenticate with the robot.");
+  }
+
+  const auto authenticate_result = robot_->Authenticate(username, password);
+  if(!authenticate_result)
+  {
+    return tl::make_unexpected("Authentication with provided username and password did not succeed.");
+  }
+
+  // Start time synchronization between the robot and the client system.
+  // This must be done only after a successful authentication.
+  const auto start_time_sync_response = robot_->StartTimeSync();
```

Comment:
Does this block until time sync is established?  If not we want to do that before returning.

Reply:
Based on my understanding of how this client works in the Spot SDK, it does block until the time sync is established. In combination with then next request to `GetTimeSyncThread` a few lines later, the clock sync will be ready to use before this function returns.